spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-sample/src/main/java/com/example/WebController.java

```diff
@@ -130,7 +130,7 @@ public RedirectView multipull(
 
 	@GetMapping("/subscribe")
 	public RedirectView subscribe(@RequestParam("subscription") String subscriptionName) {
-		Subscriber subscriber = this.pubSubTemplate.subscribe(subscriptionName, (message) -> {
+		Subscriber subscriber = this.pubSubTemplate.subscribe(subscriptionName, message -> {
```

Comment:
That comes from new project: https://github.com/spring-io/spring-javaformat, which we are utilizing through our Spring Cloud parent and there is a SpringCheck rule, which doesn't allow us to have lamdas like this.

Reply:
Do you know which rule it is so that we can add it to our `checkstyle.xml`?

Reply:
I'd rather suggest to remove our altogether and just rely on whatever is provided by our Spring Cloud parent.
This way we are going to align our code style with all the Spring Cloud portfolio.

In one of my project recently I did something like this in the root pom:
```
			<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-checkstyle-plugin</artifactId>
				<executions>
					<execution>
						<id>checkstyle-validation</id>
						<phase>validate</phase>
						<goals>
							<goal>check</goal>
						</goals>
						<configuration>
							<includeTestSourceDirectory>true</includeTestSourceDirectory>
							<consoleOutput>true</consoleOutput>
							<failsOnError>true</failsOnError>
							<failOnViolation>true</failOnViolation>
							<violationSeverity>warning</violationSeverity>
						</configuration>
					</execution>
				</executions>
                         </plugin>
```


That's all: no any Checkstyle related files in my project!

Reply:
I think we tried to use the parent configuration, and it wasn't strict enough.
For now, I would want to find the specific rule and add it because I think our current configuration is pretty close.

Reply:
@meltsufin ,

would you mind to elaborate more about "wasn't strict enough"?

I have just changed the project to rely on the parent Checkstyle config (I think that is this one: https://github.com/spring-cloud/spring-cloud-build/blob/master/spring-cloud-build-tools/src/main/resources/checkstyle.xml) and I get the report like this:
```
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\Credentials.java:25: First sentence should end with a period. [JavadocStyle]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\CredentialsSupplier.java:30:20: Javadoc element descriptions should not start with an uppercase letter. [SpringJavadoc]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\DefaultCredentialsProvider.java:81:39: Javadoc element descriptions should not start with an uppercase letter. [SpringJavadoc]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\DefaultCredentialsProvider.java:137:46: Lambda argument missing parentheses. [SpringLambda]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:25:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:26:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:27:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:28:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:29:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:30:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:31:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:32:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:33:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\GcpScope.java:34:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\UsageTrackingHeaderProvider.java:33:9: Missing a Javadoc comment. [JavadocVariable]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\main\java\org\springframework\cloud\gcp\core\UsageTrackingHeaderProvider.java:35: First sentence should end with a period. [JavadocStyle]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\test\java\org\springframework\cloud\gcp\core\DefaultCredentialsProviderTests.java:22:1: Illegal import - com.google.common.collect.ImmutableList. [IllegalImport]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\test\java\org\springframework\cloud\gcp\core\DefaultCredentialsProviderTests.java:25: Please use AssertJ imports. [RegexpSinglelineJava]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\test\java\org\springframework\cloud\gcp\core\DefaultCredentialsProviderTests.java:26: Please use AssertJ imports. [RegexpSinglelineJava]
[ERROR] C:\SpringIO\spring-cloud\spring-cloud-gcp\spring-cloud-gcp-core\src\test\java\org\springframework\cloud\gcp\core\DefaultCredentialsProviderTests.java:28: Javadoc has empty description section. [JavadocStyle]
```

What you are looking for in regards to this lambda I guess is this in that config: https://github.com/spring-cloud/spring-cloud-build/blob/master/spring-cloud-build-tools/src/main/resources/checkstyle.xml#L175

So, WDYT now?

Reply:
Thanks, it looks promising! I created #1195 to try to switch over to checkstyle in `spring-cloud-build`.
When I said we found it not strict enough, I was thinking of this recent [PR](https://github.com/spring-cloud/spring-cloud-gcp/pull/1161#pullrequestreview-169982744). I'm not sure if this is still the case though.